core/src/main/kotlin/com/uber/sdk2/core/ui/UberButton.kt

```diff
@@ -0,0 +1,257 @@
+      } else {
+        setBackgroundColor(backgroundAttributes.getColor(0, Color.BLACK))
+      }
+    } finally {
+      backgroundAttributes.recycle()
+    }
+  }
+
+  @SuppressLint("ResourceType")
+  private fun setDrawableAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources =
+      intArrayOf(
+        android.R.attr.drawableLeft,
+        android.R.attr.drawableTop,
+        android.R.attr.drawableRight,
+        android.R.attr.drawableBottom,
+        android.R.attr.drawablePadding,
+      )
+    val drawableAttributes =
+      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
+    try {
+      setCompoundDrawablesWithIntrinsicBounds(
+        drawableAttributes.getResourceId(0, 0),
+        drawableAttributes.getResourceId(1, 0),
+        drawableAttributes.getResourceId(2, 0),
+        drawableAttributes.getResourceId(3, 0),
+      )
+      compoundDrawablePadding = drawableAttributes.getDimensionPixelSize(4, 0)
+    } finally {
+      drawableAttributes.recycle()
+    }
+  }
+
+  @SuppressLint("ResourceType")
+  private fun setPaddingAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources =
+      intArrayOf(
+        android.R.attr.padding,
+        android.R.attr.paddingLeft,
+        android.R.attr.paddingTop,
+        android.R.attr.paddingRight,
+        android.R.attr.paddingBottom,
+      )
+    val paddingAttributes =
+      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
+    try {
+      val padding = paddingAttributes.getDimensionPixelOffset(0, 0)
+      var paddingLeft = paddingAttributes.getDimensionPixelSize(1, 0)
+      paddingLeft = if (paddingLeft == 0) padding else paddingLeft
+      var paddingTop = paddingAttributes.getDimensionPixelSize(2, 0)
+      paddingTop = if (paddingTop == 0) padding else paddingTop
+      var paddingRight = paddingAttributes.getDimensionPixelSize(3, 0)
+      paddingRight = if (paddingRight == 0) padding else paddingRight
+      var paddingBottom = paddingAttributes.getDimensionPixelSize(4, 0)
+      paddingBottom = if (paddingBottom == 0) padding else paddingBottom
+      setPadding(paddingLeft, paddingTop, paddingRight, paddingBottom)
+    } finally {
+      paddingAttributes.recycle()
+    }
+  }
+
+  @SuppressLint("ResourceType")
+  private fun setTextAttributes(
+    context: Context,
+    attrs: AttributeSet?,
+    defStyleAttr: Int,
+    defStyleRes: Int,
+  ) {
+    val attrsResources =
+      intArrayOf(
+        android.R.attr.textColor,
+        android.R.attr.gravity,
+        android.R.attr.textStyle,
+        android.R.attr.text,
+      )
+    val textAttributes =
+      context.theme.obtainStyledAttributes(attrs, attrsResources, defStyleAttr, defStyleRes)
+    try {
+      setTextColor(textAttributes.getColor(0, Color.WHITE))
```

Comment:
Should we add attributes for dark mode?

Reply:
That can be added in the composable button later. This button is added for backward compatibility and will be more in ktlo mode